api/v1alpha1/warehouse_types.go

```diff
@@ -267,6 +284,127 @@ type WarehouseStatus struct {
+	// ID is the identifier of the commit. This typically is a SHA-1 hash.
+	// +kubebuilder:validation:MinLength=1
+	ID string `json:"id,omitempty" protobuf:"bytes,1,opt,name=id"`
+
+	// Branch is the branch in which the commit was found. This field is
+	// optional, and populated based on the CommitSelectionStrategy of the
+	// GitSubscription.
+	// +optional
+	Branch string `json:"branch,omitempty" protobuf:"bytes,2,opt,name=branch"`
+
+	// Tag is the tag that resolved to this commit. This field is optional, and
+	// populated based on the CommitSelectionStrategy of the GitSubscription.
+	// +optional
+	Tag string `json:"tag,omitempty" protobuf:"bytes,3,opt,name=tag"`
+
+	// Subject is the subject of the commit (i.e. the first line of the commit
+	// message).
+	// +optional
+	Subject string `json:"subject,omitempty" protobuf:"bytes,4,opt,name=subject"`
+
+	// Author is the author of the commit.
+	// +optional
+	Author string `json:"author,omitempty" protobuf:"bytes,5,opt,name=author"`
+
+	// CreatedAt is the time at which the commit or annotated tag was created.
+	// +optional
+	CreatedAt *metav1.Time `json:"createdAt,omitempty" protobuf:"bytes,6,opt,name=createdAt"`
+}
+
+// ImageDiscoveryResult represents the result of an image discovery operation
+// for an ImageSubscription.
+type ImageDiscoveryResult struct {
+	// RepoURL is the repository URL of the image, as specified in the
+	// ImageSubscription.
+	// +kubebuilder:validation:MinLength=1
+	RepoURL string `json:"repoURL" protobuf:"bytes,1,opt,name=repoURL"`
+
+	// Images is a list of images discovered by the Warehouse for the
+	// ImageSubscription. An empty list indicates that the discovery operation
+	// was successful, but no images matching the ImageSubscription criteria were
+	// found.
+	// +optional
+	Images []DiscoveredImage `json:"images" protobuf:"bytes,2,rep,name=images"`
+}
+
+// DiscoveredImage represents an image discovered by a Warehouse for an
+// ImageSubscription.
+type DiscoveredImage struct {
+	// Tag is the tag of the image.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[\w.\-\_]+$`
+	Tag string `json:"tag" protobuf:"bytes,1,opt,name=tag"`
+	// Digest is the digest of the image.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:Pattern=`^[a-z0-9]+:[a-f0-9]+$`
+	Digest string `json:"digest" protobuf:"bytes,2,opt,name=digest"`
+	// Platform is the target platform of the image. This field is optional, and
+	// only populated if the ImageSubscription specifies a Platform.
+	// +optional
+	Platform string `json:"platform,omitempty" protobuf:"bytes,3,opt,name=platform"`
+	// GitRepoURL is the URL of the Git repository that contains the source
+	// code for this image. This field is optional, and only populated if the
+	// ImageSubscription specifies a GitRepoURL.
+	// +optional
+	GitRepoURL string `json:"gitRepoURL,omitempty" protobuf:"bytes,4,opt,name=gitRepoURL"`
```

Comment:
Any thoughts on this moving up a level to the `ImageDiscoveryResult`? Isn't it going to be the same for _every_ `DiscoveredImage` within a result?

Reply:
It's tag specific for GitHub URLs, due to:

```go
func getGithubImageSourceURL(gitRepoURL, tag string) string {
	return fmt.Sprintf("%s/tree/%s", git.NormalizeURL(gitRepoURL), tag)
}
```

Reply:
I'd forgotten about that.